### PR TITLE
oem-ibm: Capacity on demand licensing support

### PR DESCRIPTION
**Pull request by jaypadath**
This commit supports:
1. Creation or update of new license interfaces and properties supplied by host
2. Clearing of license status to Unknown, during host power off transition
3. Restoration of license interfaces during reset reload and setting the license status to Unknown

Tested By:
1. Verified that dbus hosting is done by pldm
2. CoD(Capacity on demand) values are coming up in the BMC
3. Verified that reboot of BMC does not clear values

Change-Id: If1018e0e789cbad290cea6bd8b2f2715dfd5a170

### CODE REVIEW

File: oem/ibm/libpldmresponder/file_io.hpp [hunk 1]
```diff
@@ -314,6 +317,40 @@ class Handler : public CmdHandler
                 }
             }
         });
+        codLicensesubs = std::make_unique<sdbusplus::bus::match::match>(
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/file_io.hpp:320]: codLicenseSubs

File: oem/ibm/libpldmresponder/file_io.hpp [hunk 2]
```diff
@@ -314,6 +317,40 @@ class Handler : public CmdHandler
...
+                    dbusToFileHandlers
+                        .emplace_back(
+                            std::make_unique<
+                                pldm::requester::oem_ibm::DbusToFileHandler>(
+                                hostSockFd, hostEid, instanceIdDb, path,
+                                handler))
+                        ->newLicFileAvailable(licenseStr);
+                    break;
+                }
+                break;
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/file_io.hpp:351]: why a second "break" statement?

File: oem/ibm/libpldmresponder/file_io_type_lic.cpp [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+        pldm::utils::PropertyValue value =
+            "com.ibm.License.LicenseManager.Status.IncorrectSequence";
+
+        try
+        {
+            pldm::utils::DBusHandler().setDbusProperty(dbusMapping, value);
+        }
+        catch (const std::exception& e)
+        {
+            error(
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/file_io_type_lic.cpp:285]: change the lg2 style

File: oem/ibm/libpldmresponder/file_io_type_lic.cpp [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+        pldm::utils::PropertyValue value =
+            "com.ibm.License.LicenseManager.Status.InvalidHostState";
+
+        try
+        {
+            pldm::utils::DBusHandler().setDbusProperty(dbusMapping, value);
+        }
+        catch (const std::exception& e)
+        {
+            error(
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/file_io_type_lic.cpp:268]: lg2 style

File: oem/ibm/libpldmresponder/file_io_type_lic.cpp [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+            "com.ibm.License.LicenseManager.Status.IncorrectSystem";
+
+        try
+        {
+            pldm::utils::DBusHandler().setDbusProperty(dbusMapping, value);
+        }
+        catch (const std::exception& e)
+        {
+            error(
+                "failed to set status property of license manager, ERROR={ERR_EXCEP}",
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/file_io_type_lic.cpp:252]: lg2 style

File: oem/ibm/libpldmresponder/file_io_type_lic.cpp [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+        pldm::utils::PropertyValue value =
+            "com.ibm.License.LicenseManager.Status.ActivationFailed";
+
+        try
+        {
+            pldm::utils::DBusHandler().setDbusProperty(dbusMapping, value);
+        }
+        catch (const std::exception& e)
+        {
+            error(
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/file_io_type_lic.cpp:234]: change the lg2 style

File: oem/ibm/libpldmresponder/file_io_type_lic.cpp [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+        pldm::utils::PropertyValue value =
+            "com.ibm.License.LicenseManager.Status.Pending";
+
+        try
+        {
+            pldm::utils::DBusHandler().setDbusProperty(dbusMapping, value);
+        }
+        catch (const std::exception& e)
+        {
+            error(
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/file_io_type_lic.cpp:217]: lg2 style

File: oem/ibm/libpldmresponder/file_io_type_lic.cpp [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+        pldm::utils::PropertyValue value =
+            "com.ibm.License.LicenseManager.Status.Activated";
+
+        try
+        {
+            pldm::utils::DBusHandler().setDbusProperty(dbusMapping, value);
+        }
+        catch (const std::exception& e)
+        {
+            error(
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/file_io_type_lic.cpp:200]: lg2 style

File: oem/ibm/libpldmresponder/file_io_type_lic.cpp [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+        pldm::utils::PropertyValue value =
+            "com.ibm.License.LicenseManager.Status.InvalidLicense";
+
+        try
+        {
+            pldm::utils::DBusHandler().setDbusProperty(dbusMapping, value);
+        }
+        catch (const std::exception& e)
+        {
+            error(
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/file_io_type_lic.cpp:183]: lg2 style

**Review by Pavithrab7 — CHANGES_REQUESTED** [review]

File: oem/ibm/libpldmresponder/file_io.hpp (hunk 1 above)
Comment (jaypadath) [reply oem/ibm/libpldmresponder/file_io.hpp:320]: Done

File: oem/ibm/libpldmresponder/file_io.hpp (hunk 2 above)
Comment (jaypadath) [reply oem/ibm/libpldmresponder/file_io.hpp:351]: Removed

File: oem/ibm/libpldmresponder/file_io_type_lic.cpp (hunk 3 above)
Comment (jaypadath) [reply oem/ibm/libpldmresponder/file_io_type_lic.cpp:285]: Done

File: oem/ibm/libpldmresponder/file_io_type_lic.cpp (hunk 4 above)
Comment (jaypadath) [reply oem/ibm/libpldmresponder/file_io_type_lic.cpp:268]: Done

File: oem/ibm/libpldmresponder/file_io_type_lic.cpp (hunk 5 above)
Comment (jaypadath) [reply oem/ibm/libpldmresponder/file_io_type_lic.cpp:252]: Done

File: oem/ibm/libpldmresponder/file_io_type_lic.cpp (hunk 6 above)
Comment (jaypadath) [reply oem/ibm/libpldmresponder/file_io_type_lic.cpp:234]: Done

File: oem/ibm/libpldmresponder/file_io_type_lic.cpp (hunk 7 above)
Comment (jaypadath) [reply oem/ibm/libpldmresponder/file_io_type_lic.cpp:217]: Done

File: oem/ibm/libpldmresponder/file_io_type_lic.cpp (hunk 8 above)
Comment (jaypadath) [reply oem/ibm/libpldmresponder/file_io_type_lic.cpp:200]: Done

File: oem/ibm/libpldmresponder/file_io_type_lic.cpp (hunk 9 above)
Comment (jaypadath) [reply oem/ibm/libpldmresponder/file_io_type_lic.cpp:183]: Done

**Review by Pavithrab7 — APPROVED** [review]

**Pull request closed without merging**